packages/@lwc/babel-plugin-component/src/decorators/index.js

```diff
@@ -137,19 +138,12 @@ function removeImportSpecifiers(specifiers) {
     }
 }
 
-function invalidDecorators() {
-    return {
-        Decorator(path) {
-            throw generateError(path.parentPath, {
-                errorInfo: DecoratorErrors.INVALID_DECORATOR_WITH_NAME,
-                messageArgs: [
-                    path.node.expression.name,
-                    LWC_DECORATORS.join(', '),
-                    LWC_PACKAGE_ALIAS,
-                ],
-            });
-        },
-    };
+function getDecoratorName(path) {
+    const { node } = path;
+    const { expression } = node;
+    const { type } = expression;
+    // Decorators can be a CallExpression if invoked as a function (e.g., @wire())
+    return type === 'Identifier' ? expression.name : expression.callee.name;
```

Comment:
As implemented by babel today, decorators can reference any expression: https://astexplorer.net/#/gist/bba34f12e529f201df8992523fd5e49f/0dd9968deb80301ae0bdbdaf6b5b40a87fb74693.

While in the simple case this method work, it will fail in more complex cases. We can either:
- report a generic error message. This should be enough since the location information should be enough.
- report 2 types of error messages: one error message when the decorator name is easy to extract, and a more generic error message in the other cases.